In [1]:
from glob import glob
import pandas as pd
from scipy.stats import mannwhitneyu, fisher_exact
from statsmodels.sandbox.stats.multicomp import multipletests
import numpy as np
import seaborn as sns
from skbio.stats.composition import clr

sns.set_style('whitegrid')

def p_adjust(pvalues, method='fdr_bh'):
    res = multipletests(pvalues, method=method)
    return np.array(res[1], dtype=float)

# Correlate proteomics data with median titers
##### 7/18/22
##### Michael Shaffer
##### Merck ESC, Sys bio group

After some success with KOs and more success with metabolomics we decided to go into proteomics. Same exact approach as in metabolomics.

## Read in the data

This is a data sheet that I got directly from Hendrik.

In [2]:
raw_proteomics = pd.read_csv('../../data/proteomics_abunds.txt', sep='\t', index_col=0).transpose()
raw_proteomics.head()

BioID Well Run Number  \
LFQ intensity 005_HFX_HW_RAW_IMC_A1_106A  106A   A1          5   
LFQ intensity 006_HFX_HW_RAW_IMC_A2_209A  209A   A2          6   
LFQ intensity 012_HFX_HW_RAW_IMC_A8_214A  214A   A8         12   
LFQ intensity 021_HFX_HW_RAW_IMC_C3_227A  227A   C3         21   
LFQ intensity 024_HFX_HW_RAW_IMC_C6_124A  124A   C6         24   

                                                Hospital VisitID Age  \
LFQ intensity 005_HFX_HW_RAW_IMC_A1_106A  Well Check 02m      V5  63   
LFQ intensity 006_HFX_HW_RAW_IMC_A2_209A  Well Check 02m      V5  55   
LFQ intensity 012_HFX_HW_RAW_IMC_A8_214A  Well Check 02m      V5  89   
LFQ intensity 021_HFX_HW_RAW_IMC_C3_227A  Well Check 02m      V5  57   
LFQ intensity 024_HFX_HW_RAW_IMC_C6_124A  Well Check 02m      V5  97   

                                               Draw Matrix      Draw_Matrix  \
LFQ intensity 005_HFX_HW_RAW_IMC_A1_106A  Heelstick  Serum  Heelstick_Serum   
LFQ intensity 006_HFX_HW_RAW_IMC_A2_209A  Heelstick  Serum  Heelstick_Serum   
LFQ intensity 012_HFX_HW_RAW_IMC_A8_214A  Heelstick  Serum  Heelstick_Serum   
LFQ intensity 021_HFX_HW_RAW_IMC_C3_227A  Heelstick  Serum  Heelstick_Serum   
LFQ intensity 024_HFX_HW_RAW_IMC_C6_124A  Heelstick  Serum  Heelstick_Serum   

                                            Draw_Matrix_Time  ...  \
LFQ intensity 005_HFX_HW_RAW_IMC_A1_106A  V5_Heelstick_Serum  ...   
LFQ intensity 006_HFX_HW_RAW_IMC_A2_209A  V5_Heelstick_Serum  ...   
LFQ intensity 012_HFX_HW_RAW_IMC_A8_214A  V5_Heelstick_Serum  ...   
LFQ intensity 021_HFX_HW_RAW_IMC_C3_227A  V5_Heelstick_Serum  ...   
LFQ intensity 024_HFX_HW_RAW_IMC_C6_124A  V5_Heelstick_Serum  ...   

                                         sp|Q6UWP8|SBSN_HUMAN  \
LFQ intensity 005_HFX_HW_RAW_IMC_A1_106A           18.2125721   
LFQ intensity 006_HFX_HW_RAW_IMC_A2_209A          16.90970612   
LFQ intensity 012_HFX_HW_RAW_IMC_A8_214A          13.34554386   
LFQ intensity 021_HFX_HW_RAW_IMC_C3_227A          20.80704689   
LFQ intensity 024_HFX_HW_RAW_IMC_C6_124A          17.50129318   

                                         sp|Q6UXB8|PI16_HUMAN  \
LFQ intensity 005_HFX_HW_RAW_IMC_A1_106A          23.48308372   
LFQ intensity 006_HFX_HW_RAW_IMC_A2_209A          23.31561852   
LFQ intensity 012_HFX_HW_RAW_IMC_A8_214A          23.58817863   
LFQ intensity 021_HFX_HW_RAW_IMC_C3_227A          23.25824928   
LFQ intensity 024_HFX_HW_RAW_IMC_C6_124A          23.05339622   

                                         sp|Q92736|RYR2_HUMAN  \
LFQ intensity 005_HFX_HW_RAW_IMC_A1_106A          20.45380974   
LFQ intensity 006_HFX_HW_RAW_IMC_A2_209A          18.60210037   
LFQ intensity 012_HFX_HW_RAW_IMC_A8_214A          18.65951729   
LFQ intensity 021_HFX_HW_RAW_IMC_C3_227A          21.18655968   
LFQ intensity 024_HFX_HW_RAW_IMC_C6_124A          21.60560417   

                                         sp|Q96IY4|CBPB2_HUMAN  \
LFQ intensity 005_HFX_HW_RAW_IMC_A1_106A             21.819561   
LFQ intensity 006_HFX_HW_RAW_IMC_A2_209A           22.21127701   
LFQ intensity 012_HFX_HW_RAW_IMC_A8_214A            21.8640461   
LFQ intensity 021_HFX_HW_RAW_IMC_C3_227A           21.24667168   
LFQ intensity 024_HFX_HW_RAW_IMC_C6_124A           21.72242355   

                                         sp|Q96PD5|PGRP2_HUMAN  \
LFQ intensity 005_HFX_HW_RAW_IMC_A1_106A           25.08229065   
LFQ intensity 006_HFX_HW_RAW_IMC_A2_209A           24.88887978   
LFQ intensity 012_HFX_HW_RAW_IMC_A8_214A           25.16353798   
LFQ intensity 021_HFX_HW_RAW_IMC_C3_227A           25.07989311   
LFQ intensity 024_HFX_HW_RAW_IMC_C6_124A           25.20258331   

                                         sp|Q9HDC9|APMAP_HUMAN  \
LFQ intensity 005_HFX_HW_RAW_IMC_A1_106A           20.85553551   
LFQ intensity 006_HFX_HW_RAW_IMC_A2_209A           20.59141159   
LFQ intensity 012_HFX_HW_RAW_IMC_A8_214A           21.04162788   
LFQ intensity 021_HFX_HW_RAW_IMC_C3_227A           19.46746445   
LFQ intensity 024_HFX_HW

This is removing all columns that doen't have abundance information, taking only unique ID from row names and converting to floats.

In [3]:
proteomics = raw_proteomics.loc[[i for i in raw_proteomics.index if i.startswith('LFQ intensity')],
                                raw_proteomics.columns[list(raw_proteomics.columns).index('Subject ID')+1:]]
proteomics.index = [i.split()[-1] for i in proteomics.index]
proteomics = proteomics.astype(float)
proteomics.head()

CON__P00761  sp|P02533|K1C14_HUMAN;CON__P02533  \
005_HFX_HW_RAW_IMC_A1_106A    29.414034                          19.480482   
006_HFX_HW_RAW_IMC_A2_209A    29.336454                          19.774490   
012_HFX_HW_RAW_IMC_A8_214A    28.906614                          19.529310   
021_HFX_HW_RAW_IMC_C3_227A    29.769306                          21.061234   
024_HFX_HW_RAW_IMC_C6_124A    29.567360                          20.363079   

                            sp|P02768|ALBU_HUMAN;CON__P02768-1  CON__P02769  \
005_HFX_HW_RAW_IMC_A1_106A                           33.365807    23.197741   
006_HFX_HW_RAW_IMC_A2_209A                           32.956917    23.397673   
012_HFX_HW_RAW_IMC_A8_214A                           33.174339    24.401390   
021_HFX_HW_RAW_IMC_C3_227A                           33.328308    24.742115   
024_HFX_HW_RAW_IMC_C6_124A                           32.930107    23.502518   

                            sp|P48668|K2C6C_HUMAN;CON__P48668;CON__P04259  \
005_HFX_HW_RAW_IMC_A1_106A                                      21.572432   
006_HFX_HW_RAW_IMC_A2_209A                                      21.685078   
012_HFX_HW_RAW_IMC_A8_214A                                      19.594507   
021_HFX_HW_RAW_IMC_C3_227A                                      20.643036   
024_HFX_HW_RAW_IMC_C6_124A                                      20.780497   

                            sp|P13645|K1C10_HUMAN;CON__P13645  \
005_HFX_HW_RAW_IMC_A1_106A                          21.094681   
006_HFX_HW_RAW_IMC_A2_209A                          22.188005   
012_HFX_HW_RAW_IMC_A8_214A                          22.131947   
021_HFX_HW_RAW_IMC_C3_227A                          20.368902   
024_HFX_HW_RAW_IMC_C6_124A                          19.634420   

                            sp|P13647|K2C5_HUMAN;CON__P13647  \
005_HFX_HW_RAW_IMC_A1_106A                         19.310707   
006_HFX_HW_RAW_IMC_A2_209A                         20.725851   
012_HFX_HW_RAW_IMC_A8_214A                         19.528219   
021_HFX_HW_RAW_IMC_C3_227A                         20.235897   
024_HFX_HW_RAW_IMC_C6_124A                         20.836931   

                            CON__P35908;CON__P35908v2;sp|P35908|K22E_HUMAN  \
005_HFX_HW_RAW_IMC_A1_106A                                       19.288029   
006_HFX_HW_RAW_IMC_A2_209A                                       20.871471   
012_HFX_HW_RAW_IMC_A8_214A                                       20.954256   
021_HFX_HW_RAW_IMC_C3_227A                                       17.218355   
024_HFX_HW_RAW_IMC_C6_124A                                       17.906363   

                            sp|Q86YZ3|HORN_HUMAN;CON__Q86YZ3  \
005_HFX_HW_RAW_IMC_A1_106A                         19.483946   
006_HFX_HW_RAW_IMC_A2_209A                         19.841579   
012_HFX_HW_RAW_IMC_A8_214A                         18.534006   
021_HFX_HW_RAW_IMC_C3_227A                         19.047693   
024_HFX_HW_RAW_IMC_C6_124A                         18.741560   

                            sp|A0A075B6H9|LV469_HUMAN  ...  \
005_HFX_HW_RAW_IMC_A1_106A                  21.265715  ...   
006_HFX_HW_RAW_IMC_A2_209A                  20.858858  ...   
012_HFX_HW_RAW_IMC_A8_214A                  21.559875  ...   
021_HFX_HW_RAW_IMC_C3_227A                  21.543921  ...   
024_HFX_HW_RAW_IMC_C6_124A                  22.351000  ...   

                            sp|Q6UWP8|SBSN_HUMAN  sp|Q6UXB8|PI16_HUMAN  \
005_HFX_HW_RAW_IMC_A1_106A             18.212572             23.483084   
006_HFX_HW_RAW_IMC_A2_209A             16.909706             23.315619   
012_HFX_HW_RAW_IMC_A8_214A             13.345544             23.588179   
021_HFX_HW_RAW_IMC_C3_227A             20.807047             23.258249   
024_HFX_HW_RAW_IMC_C6_124A             17.501293             23.053396   

                            sp|Q92736|RYR2_HUMAN  sp|Q96IY4|CBPB2_HUMAN  \
005_HFX_HW_RAW_IMC_A1_106A             20.453810              21.819561   
006_HFX_HW_RAW_IMC_A2_2

In [4]:
print(len(proteomics.columns))

250


250 proteins measured.

Convert the abundances to relative abundance for comparison of normalizations.

In [5]:
proteomics_rel = proteomics.div(proteomics.sum(axis=1), axis=0)
proteomics_rel.head()

CON__P00761  sp|P02533|K1C14_HUMAN;CON__P02533  \
005_HFX_HW_RAW_IMC_A1_106A     0.005043                           0.003340   
006_HFX_HW_RAW_IMC_A2_209A     0.005066                           0.003415   
012_HFX_HW_RAW_IMC_A8_214A     0.005009                           0.003384   
021_HFX_HW_RAW_IMC_C3_227A     0.005104                           0.003611   
024_HFX_HW_RAW_IMC_C6_124A     0.005069                           0.003491   

                            sp|P02768|ALBU_HUMAN;CON__P02768-1  CON__P02769  \
005_HFX_HW_RAW_IMC_A1_106A                            0.005721     0.003977   
006_HFX_HW_RAW_IMC_A2_209A                            0.005691     0.004041   
012_HFX_HW_RAW_IMC_A8_214A                            0.005749     0.004229   
021_HFX_HW_RAW_IMC_C3_227A                            0.005715     0.004242   
024_HFX_HW_RAW_IMC_C6_124A                            0.005646     0.004030   

                            sp|P48668|K2C6C_HUMAN;CON__P48668;CON__P04259  \
005_HFX_HW_RAW_IMC_A1_106A                                       0.003699   
006_HFX_HW_RAW_IMC_A2_209A                                       0.003745   
012_HFX_HW_RAW_IMC_A8_214A                                       0.003396   
021_HFX_HW_RAW_IMC_C3_227A                                       0.003540   
024_HFX_HW_RAW_IMC_C6_124A                                       0.003563   

                            sp|P13645|K1C10_HUMAN;CON__P13645  \
005_HFX_HW_RAW_IMC_A1_106A                           0.003617   
006_HFX_HW_RAW_IMC_A2_209A                           0.003832   
012_HFX_HW_RAW_IMC_A8_214A                           0.003835   
021_HFX_HW_RAW_IMC_C3_227A                           0.003493   
024_HFX_HW_RAW_IMC_C6_124A                           0.003366   

                            sp|P13647|K2C5_HUMAN;CON__P13647  \
005_HFX_HW_RAW_IMC_A1_106A                          0.003311   
006_HFX_HW_RAW_IMC_A2_209A                          0.003579   
012_HFX_HW_RAW_IMC_A8_214A                          0.003384   
021_HFX_HW_RAW_IMC_C3_227A                          0.003470   
024_HFX_HW_RAW_IMC_C6_124A                          0.003573   

                            CON__P35908;CON__P35908v2;sp|P35908|K22E_HUMAN  \
005_HFX_HW_RAW_IMC_A1_106A                                        0.003307   
006_HFX_HW_RAW_IMC_A2_209A                                        0.003604   
012_HFX_HW_RAW_IMC_A8_214A                                        0.003631   
021_HFX_HW_RAW_IMC_C3_227A                                        0.002952   
024_HFX_HW_RAW_IMC_C6_124A                                        0.003070   

                            sp|Q86YZ3|HORN_HUMAN;CON__Q86YZ3  \
005_HFX_HW_RAW_IMC_A1_106A                          0.003341   
006_HFX_HW_RAW_IMC_A2_209A                          0.003426   
012_HFX_HW_RAW_IMC_A8_214A                          0.003212   
021_HFX_HW_RAW_IMC_C3_227A                          0.003266   
024_HFX_HW_RAW_IMC_C6_124A                          0.003213   

                            sp|A0A075B6H9|LV469_HUMAN  ...  \
005_HFX_HW_RAW_IMC_A1_106A                   0.003646  ...   
006_HFX_HW_RAW_IMC_A2_209A                   0.003602  ...   
012_HFX_HW_RAW_IMC_A8_214A                   0.003736  ...   
021_HFX_HW_RAW_IMC_C3_227A                   0.003694  ...   
024_HFX_HW_RAW_IMC_C6_124A                   0.003832  ...   

                            sp|Q6UWP8|SBSN_HUMAN  sp|Q6UXB8|PI16_HUMAN  \
005_HFX_HW_RAW_IMC_A1_106A              0.003123              0.004026   
006_HFX_HW_RAW_IMC_A2_209A              0.002920              0.004026   
012_HFX_HW_RAW_IMC_A8_214A              0.002313              0.004088   
021_HFX_HW_RAW_IMC_C3_227A              0.003568              0.003988   
024_HFX_HW_RAW_IMC_C6_124A              0.003001              0.003953   

                            sp|Q92736|RYR2_HUMAN  sp|Q96IY4|CBPB2_HUMAN  \
005_HFX_HW_RAW_IMC_A1_106A              0.003507               0.003741   
006_HFX_HW_RAW_IMC_A2_2

Same thing but with CLR.

In [6]:
proteomics_clr = pd.DataFrame(clr(proteomics + .001), index=proteomics.index, columns=proteomics.columns)
proteomics_clr.head()

CON__P00761  sp|P02533|K1C14_HUMAN;CON__P02533  \
005_HFX_HW_RAW_IMC_A1_106A     0.239511                          -0.172531   
006_HFX_HW_RAW_IMC_A2_209A     0.244758                          -0.149664   
012_HFX_HW_RAW_IMC_A8_214A     0.234427                          -0.157710   
021_HFX_HW_RAW_IMC_C3_227A     0.251985                          -0.094045   
024_HFX_HW_RAW_IMC_C6_124A     0.245111                          -0.127821   

                            sp|P02768|ALBU_HUMAN;CON__P02768-1  CON__P02769  \
005_HFX_HW_RAW_IMC_A1_106A                            0.365567     0.002103   
006_HFX_HW_RAW_IMC_A2_209A                            0.361124     0.018572   
012_HFX_HW_RAW_IMC_A8_214A                            0.372129     0.065003   
021_HFX_HW_RAW_IMC_C3_227A                            0.364910     0.067021   
024_HFX_HW_RAW_IMC_C6_124A                            0.352824     0.015557   

                            sp|P48668|K2C6C_HUMAN;CON__P48668;CON__P04259  \
005_HFX_HW_RAW_IMC_A1_106A                                      -0.070532   
006_HFX_HW_RAW_IMC_A2_209A                                      -0.057437   
012_HFX_HW_RAW_IMC_A8_214A                                      -0.154378   
021_HFX_HW_RAW_IMC_C3_227A                                      -0.114100   
024_HFX_HW_RAW_IMC_C6_124A                                      -0.107531   

                            sp|P13645|K1C10_HUMAN;CON__P13645  \
005_HFX_HW_RAW_IMC_A1_106A                          -0.092927   
006_HFX_HW_RAW_IMC_A2_209A                          -0.034510   
012_HFX_HW_RAW_IMC_A8_214A                          -0.032611   
021_HFX_HW_RAW_IMC_C3_227A                          -0.127468   
024_HFX_HW_RAW_IMC_C6_124A                          -0.164258   

                            sp|P13647|K2C5_HUMAN;CON__P13647  \
005_HFX_HW_RAW_IMC_A1_106A                         -0.181283   
006_HFX_HW_RAW_IMC_A2_209A                         -0.102677   
012_HFX_HW_RAW_IMC_A8_214A                         -0.157766   
021_HFX_HW_RAW_IMC_C3_227A                         -0.134019   
024_HFX_HW_RAW_IMC_C6_124A                         -0.104819   

                            CON__P35908;CON__P35908v2;sp|P35908|K22E_HUMAN  \
005_HFX_HW_RAW_IMC_A1_106A                                       -0.182458   
006_HFX_HW_RAW_IMC_A2_209A                                       -0.095676   
012_HFX_HW_RAW_IMC_A8_214A                                       -0.087288   
021_HFX_HW_RAW_IMC_C3_227A                                       -0.295493   
024_HFX_HW_RAW_IMC_C6_124A                                       -0.256382   

                            sp|Q86YZ3|HORN_HUMAN;CON__Q86YZ3  \
005_HFX_HW_RAW_IMC_A1_106A                         -0.172353   
006_HFX_HW_RAW_IMC_A2_209A                         -0.146277   
012_HFX_HW_RAW_IMC_A8_214A                         -0.210017   
021_HFX_HW_RAW_IMC_C3_227A                         -0.194528   
024_HFX_HW_RAW_IMC_C6_124A                         -0.210797   

                            sp|A0A075B6H9|LV469_HUMAN  ...  \
005_HFX_HW_RAW_IMC_A1_106A                  -0.084852  ...   
006_HFX_HW_RAW_IMC_A2_209A                  -0.096280  ...   
012_HFX_HW_RAW_IMC_A8_214A                  -0.058797  ...   
021_HFX_HW_RAW_IMC_C3_227A                  -0.071387  ...   
024_HFX_HW_RAW_IMC_C6_124A                  -0.034678  ...   

                            sp|Q6UWP8|SBSN_HUMAN  sp|Q6UXB8|PI16_HUMAN  \
005_HFX_HW_RAW_IMC_A1_106A             -0.239828              0.014328   
006_HFX_HW_RAW_IMC_A2_209A             -0.306160              0.015059   
012_HFX_HW_RAW_IMC_A8_214A             -0.538420              0.031110   
021_HFX_HW_RAW_IMC_C3_227A             -0.106187              0.005176   
024_HFX_HW_RAW_IMC_C6_124A             -0.279262             -0.003737   

                            sp|Q92736|RYR2_HUMAN  sp|Q96IY4|CBPB2_HUMAN  \
005_HFX_HW_RAW_IMC_A1_106A             -0.123777              -0.059142   
006_HFX_HW_RAW_IMC_A2_2

Now we will pull the metadata from the non-abundance columns.

In [7]:
meta_base = raw_proteomics.loc[[i for i in raw_proteomics.index if i.startswith('LFQ intensity')],
                               raw_proteomics.columns[:list(raw_proteomics.columns).index('Subject ID')+1]]
meta_base.index = [i.split()[-1] for i in meta_base.index]
column_name_replacements = {'Subject ID': 'BabyN', 'VisitID': 'VisitCode'}
meta_base.columns = [i if i not in column_name_replacements else column_name_replacements[i] for i in meta_base.columns]
meta_base['BabyN'] = [int(i) for i in meta_base['BabyN']]
meta_base.head()

BioID Well Run Number        Hospital VisitCode  \
005_HFX_HW_RAW_IMC_A1_106A  106A   A1          5  Well Check 02m        V5   
006_HFX_HW_RAW_IMC_A2_209A  209A   A2          6  Well Check 02m        V5   
012_HFX_HW_RAW_IMC_A8_214A  214A   A8         12  Well Check 02m        V5   
021_HFX_HW_RAW_IMC_C3_227A  227A   C3         21  Well Check 02m        V5   
024_HFX_HW_RAW_IMC_C6_124A  124A   C6         24  Well Check 02m        V5   

                           Age       Draw Matrix      Draw_Matrix  \
005_HFX_HW_RAW_IMC_A1_106A  63  Heelstick  Serum  Heelstick_Serum   
006_HFX_HW_RAW_IMC_A2_209A  55  Heelstick  Serum  Heelstick_Serum   
012_HFX_HW_RAW_IMC_A8_214A  89  Heelstick  Serum  Heelstick_Serum   
021_HFX_HW_RAW_IMC_C3_227A  57  Heelstick  Serum  Heelstick_Serum   
024_HFX_HW_RAW_IMC_C6_124A  97  Heelstick  Serum  Heelstick_Serum   

                              Draw_Matrix_Time Plate Responder Status  \
005_HFX_HW_RAW_IMC_A1_106A  V5_Heelstick_Serum     1              LVR   
006_HFX_HW_RAW_IMC_A2_209A  V5_Heelstick_Serum     1              NVR   
012_HFX_HW_RAW_IMC_A8_214A  V5_Heelstick_Serum     1              HVR   
021_HFX_HW_RAW_IMC_C3_227A  V5_Heelstick_Serum     1              HVR   
024_HFX_HW_RAW_IMC_C6_124A  V5_Heelstick_Serum     1              NaN   

                                          Group_ID      Group_ID_Responder  \
005_HFX_HW_RAW_IMC_A1_106A  V5_Heelstick_Serum_LVR  V5_Heelstick_Serum_LVR   
006_HFX_HW_RAW_IMC_A2_209A  V5_Heelstick_Serum_NVR  V5_Heelstick_Serum_NVR   
012_HFX_HW_RAW_IMC_A8_214A  V5_Heelstick_Serum_HVR  V5_Heelstick_Serum_HVR   
021_HFX_HW_RAW_IMC_C3_227A  V5_Heelstick_Serum_HVR  V5_Heelstick_Serum_HVR   
024_HFX_HW_RAW_IMC_C6_124A   V5_Heelstick_Serum_NA   V5_Heelstick_Serum_NA   

                           Responder Status NVRHVR Group_ID_Responder2Groups  \
005_HFX_HW_RAW_IMC_A1_106A                     NVR    V5_Heelstick_Serum_NVR   
006_HFX_HW_RAW_IMC_A2_209A                     NVR    V5_Heelstick_Serum_NVR   
012_HFX_HW_RAW_IMC_A8_214A                     HVR    V5_Heelstick_Serum_HVR   
021_HFX_HW_RAW_IMC_C3_227A                     HVR    V5_Heelstick_Serum_HVR   
024_HFX_HW_RAW_IMC_C6_124A                     NaN     V5_Heelstick_Serum_NA   

                            BabyN  
005_HFX_HW_RAW_IMC_A1_106A    106  
006_HFX_HW_RAW_IMC_A2_209A    209  
012_HFX_HW_RAW_IMC_A8_214A    214  
021_HFX_HW_RAW_IMC_C3_227A    226  
024_HFX_HW_RAW_IMC_C6_124A    124

Bring in titer data.

In [8]:
titer_data = pd.read_csv('../../data/vaccine_response/vaccine_response_y1.tsv', sep='\t', index_col=0)
titer_data.index = [int(i.split('Baby')[-1]) for i in titer_data.index]
titer_data.head()

PT   Dip   FHA   PRN   TET  PRP (Hib)  PCV ST1  PCV ST3  PCV ST4  \
106   2.5  0.21  11.0   2.5  0.30       0.39    141.0     35.0     56.0   
107   2.5  0.44   3.0   9.0  0.52       1.60   2430.0    415.0    194.0   
108   2.5  0.05   1.5   2.5  0.05       0.27     21.0      3.0     24.0   
109  27.0   NaN   NaN  63.0  1.35       7.02      NaN      NaN      NaN   
110  14.0  0.24  15.0  20.0  2.45        NaN    301.0     63.0    400.0   

     PCV ST5  ...  median_mmNorm  median_mmNorm_DTAPHib  median_mmNorm_PCV  \
106    139.0  ...       0.061955               0.052874           0.061955   
107    332.0  ...       0.449483               0.114018           0.958142   
108     41.0  ...       0.000000               0.000000           0.003102   
109      NaN  ...       0.700925               0.763049           0.486810   
110    289.0  ...       0.266219               0.284211           0.245121   

     PT_protected  Dip_protected  FHA_protected  PRN_protected  TET_protected  \
106         False           True           True          False           True   
107         False           True          False           True           True   
108         False          False          False          False          False   
109          True          False          False           True           True   
110          True           True           True           True           True   

     PRP (Hib)_protected  VR_group  
106                 True       NVR  
107                 True       NVR  
108                 True       LVR  
109                 True       NVR  
110                False       NVR  

[5 rows x 48 columns]

Split titer data to be per sample that we have in the proteomics metadata.

In [9]:
per_sample_titer_data = pd.DataFrame({sample: titer_data.loc[i] for sample, i in meta_base['BabyN'].iteritems() if i in titer_data.index}).transpose()
per_sample_titer_data.head()

PT   Dip   FHA   PRN   TET PRP (Hib) PCV ST1  \
005_HFX_HW_RAW_IMC_A1_106A   2.5  0.21  11.0   2.5   0.3      0.39   141.0   
006_HFX_HW_RAW_IMC_A2_209A   5.0  0.29   7.0   5.0  0.27      1.17   154.0   
012_HFX_HW_RAW_IMC_A8_214A  12.0  0.28  29.0  19.0  0.66       4.7     NaN   
021_HFX_HW_RAW_IMC_C3_227A   6.0  0.25  11.0  20.0  0.82      2.25   381.0   
026_HFX_HW_RAW_IMC_C8_208A   6.0  0.32   9.0   2.5  0.14      0.91   297.0   

                           PCV ST3 PCV ST4 PCV ST5  ... median_mmNorm  \
005_HFX_HW_RAW_IMC_A1_106A    35.0    56.0   139.0  ...      0.061955   
006_HFX_HW_RAW_IMC_A2_209A    40.0    89.0   679.0  ...      0.102041   
012_HFX_HW_RAW_IMC_A8_214A     NaN     NaN     NaN  ...      0.310899   
021_HFX_HW_RAW_IMC_C3_227A    94.0    90.0   322.0  ...      0.191269   
026_HFX_HW_RAW_IMC_C8_208A   193.0   181.0   278.0  ...      0.125416   

                           median_mmNorm_DTAPHib median_mmNorm_PCV  \
005_HFX_HW_RAW_IMC_A1_106A              0.052874          0.061955   
006_HFX_HW_RAW_IMC_A2_209A              0.105087          0.074972   
012_HFX_HW_RAW_IMC_A8_214A              0.310899               NaN   
021_HFX_HW_RAW_IMC_C3_227A              0.205742          0.139576   
026_HFX_HW_RAW_IMC_C8_208A              0.113055          0.125416   

                           PT_protected Dip_protected FHA_protected  \
005_HFX_HW_RAW_IMC_A1_106A        False          True          True   
006_HFX_HW_RAW_IMC_A2_209A        False          True         False   
012_HFX_HW_RAW_IMC_A8_214A         True          True          True   
021_HFX_HW_RAW_IMC_C3_227A        False          True          True   
026_HFX_HW_RAW_IMC_C8_208A        False          True          True   

                           PRN_protected TET_protected PRP (Hib)_protected  \
005_HFX_HW_RAW_IMC_A1_106A         False          True                True   
006_HFX_HW_RAW_IMC_A2_209A         False          True                True   
012_HFX_HW_RAW_IMC_A8_214A          True          True                True   
021_HFX_HW_RAW_IMC_C3_227A          True          True                True   
026_HFX_HW_RAW_IMC_C8_208A         False          True                True   

                           VR_group  
005_HFX_HW_RAW_IMC_A1_106A      NVR  
006_HFX_HW_RAW_IMC_A2_209A      NVR  
012_HFX_HW_RAW_IMC_A8_214A      NVR  
021_HFX_HW_RAW_IMC_C3_227A      NVR  
026_HFX_HW_RAW_IMC_C8_208A      NVR  

[5 rows x 48 columns]

Merge proteomics metadata and titer data. Remove samples without a VR group.

In [10]:
meta = pd.concat([meta_base, per_sample_titer_data], axis=1)
meta = meta.loc[~pd.isna(meta['VR_group'])]
meta.head()

BioID Well Run Number        Hospital VisitCode  \
005_HFX_HW_RAW_IMC_A1_106A  106A   A1          5  Well Check 02m        V5   
006_HFX_HW_RAW_IMC_A2_209A  209A   A2          6  Well Check 02m        V5   
012_HFX_HW_RAW_IMC_A8_214A  214A   A8         12  Well Check 02m        V5   
021_HFX_HW_RAW_IMC_C3_227A  227A   C3         21  Well Check 02m        V5   
026_HFX_HW_RAW_IMC_C8_208A  208A   C8         26  Well Check 02m        V5   

                           Age       Draw Matrix      Draw_Matrix  \
005_HFX_HW_RAW_IMC_A1_106A  63  Heelstick  Serum  Heelstick_Serum   
006_HFX_HW_RAW_IMC_A2_209A  55  Heelstick  Serum  Heelstick_Serum   
012_HFX_HW_RAW_IMC_A8_214A  89  Heelstick  Serum  Heelstick_Serum   
021_HFX_HW_RAW_IMC_C3_227A  57  Heelstick  Serum  Heelstick_Serum   
026_HFX_HW_RAW_IMC_C8_208A  63  Heelstick  Serum  Heelstick_Serum   

                              Draw_Matrix_Time  ... median_mmNorm  \
005_HFX_HW_RAW_IMC_A1_106A  V5_Heelstick_Serum  ...      0.061955   
006_HFX_HW_RAW_IMC_A2_209A  V5_Heelstick_Serum  ...      0.102041   
012_HFX_HW_RAW_IMC_A8_214A  V5_Heelstick_Serum  ...      0.310899   
021_HFX_HW_RAW_IMC_C3_227A  V5_Heelstick_Serum  ...      0.191269   
026_HFX_HW_RAW_IMC_C8_208A  V5_Heelstick_Serum  ...      0.125416   

                           median_mmNorm_DTAPHib median_mmNorm_PCV  \
005_HFX_HW_RAW_IMC_A1_106A              0.052874          0.061955   
006_HFX_HW_RAW_IMC_A2_209A              0.105087          0.074972   
012_HFX_HW_RAW_IMC_A8_214A              0.310899               NaN   
021_HFX_HW_RAW_IMC_C3_227A              0.205742          0.139576   
026_HFX_HW_RAW_IMC_C8_208A              0.113055          0.125416   

                           PT_protected Dip_protected FHA_protected  \
005_HFX_HW_RAW_IMC_A1_106A        False          True          True   
006_HFX_HW_RAW_IMC_A2_209A        False          True         False   
012_HFX_HW_RAW_IMC_A8_214A         True          True          True   
021_HFX_HW_RAW_IMC_C3_227A        False          True          True   
026_HFX_HW_RAW_IMC_C8_208A        False          True          True   

                            PRN_protected TET_protected PRP (Hib)_protected  \
005_HFX_HW_RAW_IMC_A1_106A          False          True                True   
006_HFX_HW_RAW_IMC_A2_209A          False          True                True   
012_HFX_HW_RAW_IMC_A8_214A           True          True                True   
021_HFX_HW_RAW_IMC_C3_227A           True          True                True   
026_HFX_HW_RAW_IMC_C8_208A          False          True                True   

                           VR_group  
005_HFX_HW_RAW_IMC_A1_106A      NVR  
006_HFX_HW_RAW_IMC_A2_209A      NVR  
012_HFX_HW_RAW_IMC_A8_214A      NVR  
021_HFX_HW_RAW_IMC_C3_227A      NVR  
026_HFX_HW_RAW_IMC_C8_208A      NVR  

[5 rows x 65 columns]

In [11]:
in_both = list(set(meta.index) & set(proteomics.index))
meta_matched = meta.loc[in_both]
print(meta_matched.shape)

(47, 65)


In [12]:
meta_matched['VisitCode'].value_counts()

V5    45
V6     2
Name: VisitCode, dtype: int64

47 samples are in the metadata and proteomics data. Of those two are V6. We will filter those out.

In [13]:
meta_v5 = meta_matched.query("VisitCode == 'V5'")

## Do correlation with un-normalized data

We will do our correlations between proteomics and median titer. I used Spearman correlation and Benjamini-Hochberg for FDR correction per set of tests.

In [14]:
proteomics_v5 = proteomics.loc[meta_v5.index]
proteomics_stats_v5_rows = list()
for protein, row in proteomics_v5.transpose().iterrows():
    lvr_abunds = row[meta_v5.query('VR_group == "LVR"').index]
    nvr_abunds = row[meta_v5.query('VR_group == "NVR"').index]
    # check for not all zeros
    # lvr_gt_20 = (lvr_abunds != 0).sum()/len(lvr_abunds) > .2
    # nvr_gt_20 = (nvr_abunds != 0).sum()/len(nvr_abunds) > .2
    lvr_gt_20 = lvr_abunds.sum()/len(lvr_abunds) > 10
    nvr_gt_20 = nvr_abunds.sum()/len(nvr_abunds) > 10
    if lvr_gt_20 or nvr_gt_20:
        stat, p_value = mannwhitneyu(lvr_abunds, nvr_abunds)
        proteomics_stats_v5_rows.append([protein, lvr_abunds.mean(), nvr_abunds.mean(), stat, p_value])
proteomics_stats_v5 = pd.DataFrame(proteomics_stats_v5_rows, columns=['protein', 'LVR_mean', 'NVR_mean', 'statistic', 'p_value']).sort_values('p_value')
proteomics_stats_v5['p_adj'] = p_adjust(proteomics_stats_v5['p_value'])
proteomics_stats_v5.head()

protein   LVR_mean   NVR_mean  statistic   p_value  \
111   sp|P02750|A2GL_HUMAN  23.149189  22.660771      247.0  0.002204   
98   sp|P02652|APOA2_HUMAN  27.290325  26.795589      242.0  0.003944   
142   sp|P05156|CFAI_HUMAN  23.295402  23.556167       58.0  0.006083   
0              CON__P00761  29.807311  29.472792      231.0  0.012217   
164   sp|P08697|A2AP_HUMAN  25.159493  25.013711      231.0  0.012217   

        p_adj  
111  0.492957  
98   0.492957  
142  0.506935  
0    0.610837  
164  0.610837